pages/recipe/[id]/index.js

```diff
@@ -0,0 +1,162 @@
+import IconButton from "@/components/Styled/IconButton";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import styled from "styled-components";
+import Link from "next/link";
+import { useState } from "react";
+
+export default function DetailPage() {
+  const [content, setContent] = useState("instructions");
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { data: recipe, isLoading, error } = useSWR(`/api/recipes/${id}`);
+
+  if (error) {
+    return <h1>error</h1>;
+  }
+  if (isLoading) {
+    return <h1>loading recipe...</h1>;
+  }
+  if (!recipe) {
+    return <h1>loading recipe...</h1>;
+  }
+
+  const {
+    _id,
+    title,
+    instructions,
+    imageLink,
+    tags,
+    youtubeLink,
+    ingredients,
+    duration,
+    difficulty,
+  } = recipe;
+
+  difficulty.toUpperCase();
+
+  return (
+    <Wrapper>
+      <IconButton
+        onClick={() => {
+          router.back();
+        }}
+        style={"arrowLeft"}
+      />
+      <StyledImage
+        src={imageLink}
+        width={400}
+        height={300}
+        alt={`recipe Image ${title}`}
+      />
+      <StyledArticle>
+        <h1>{title}</h1>
+        <p>
+          {duration} MIN | {difficulty}
+        </p>
+        <Styledh2>ingredients</Styledh2>
+        <StyledList>
+          {ingredients.map((ingredient) => (
+            <StyledListItem key={_id}>
+              <StyledP>{ingredient.name}</StyledP>
+              <StyledP>
+                {ingredient.quantity} {ingredient.unit}
+              </StyledP>
+            </StyledListItem>
+          ))}
+        </StyledList>
+        <StyledHyper>
+          <StyledLink
+            href="#instructions"
+            onClick={() => setContent("instructions")}
+          >
+            instructions
+          </StyledLink>
+          <StyledLink href="#video" onClick={() => setContent("video")}>
+            video
+          </StyledLink>
```

Comment:
You are not really navigation between pages here on between different parts of the same page.

So: Why are those elements Links and not just buttons?